### PR TITLE
ISSUE-88 Email submission simulation

### PR DESCRIPTION
**Pull request by LanKhuat**

### CODE REVIEW

File: src/main/scala-2.12/org/apache/james/gatling/jmap/RealUsageScenario.scala [hunk 1]
```diff
@@ -27,3 +27,7 @@ case object ListMessageBodyStructure extends RealUsageScenario {
 case object MessageFlagUpdates extends RealUsageScenario {
   def name = "MessageFlagUpdates"
 }
+
+case object MessageSubmission extends RealUsageScenario {
+  def name = "MessageSubmission"
```
Comment (chibenwa) [inline src/main/scala-2.12/org/apache/james/gatling/jmap/RealUsageScenario.scala]: EmailSubmission

File: src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/JmapEmail.scala [hunk 2]
```diff
@@ -114,4 +118,59 @@ object JmapEmail {
            |    "c1"]]
            |}""".stripMargin))
   }
+
+  def sendMessages(recipientFeeder: RecipientFeederBuilder): ChainBuilder = {
```
Comment (chibenwa) [inline src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/JmapEmail.scala]: sendEmail

File: src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/JmapEmail.scala [hunk 3]
```diff
@@ -114,4 +118,59 @@ object JmapEmail {
...
+      Map(messageIdSessionParam -> MessageId().id,
+        subjectSessionParam -> Subject().subject,
+        textBodySessionParam -> TextBody().text))
+
+    feed(mailFeeder)
+      .feed(recipientFeeder)
+      .exec(sendMessage())
+  }
+
+  def sendMessage(title: RequestTitle = RequestTitle("sendMessages"),
```
Comment (chibenwa) [inline src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/JmapEmail.scala]: Idem sendEmail. Even better submitEmail...

File: src/test/scala-2.12/org/apache/james/gatling/simulation/SimulationOnMailCorpus.scala [hunk 4]
```diff
@@ -26,6 +27,7 @@ trait SimulationOnMailCorpus {
     .map(new MailboxName(_))
 
   protected val feeder: SourceFeederBuilder[String] = UserFeeder.toFeeder(getUsers).circular
+  protected val recipientFeeder: RecipientFeederBuilder = RecipientFeeder.usersToFeeder(getUsers)
```
Comment (chibenwa) [inline src/test/scala-2.12/org/apache/james/gatling/simulation/SimulationOnMailCorpus.scala]: Why cant we reuse feeder like on draft?

File: src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/scenari/EmailSubmissionScenario.scala [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+class EmailSubmissionScenario {
+
+  def generate(duration: Duration, userFeeder: UserFeederBuilder, recipientFeeder: RecipientFeederBuilder): ScenarioBuilder =
+    scenario("EmailSubmissionScenario")
+      .feed(userFeeder)
+      .exec(CommonSteps.provisionSystemMailboxes())
+      .exec(retrieveAccountId)
+      .during(duration) {
+        exec(JmapEmail.sendMessages(recipientFeeder))
```
Comment (Arsnael) [inline src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/scenari/EmailSubmissionScenario.scala]: Missing checks :)

File: src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/JmapEmail.scala [hunk 6]
```diff
@@ -114,4 +118,59 @@ object JmapEmail {
...
+           |         "$SubmissionId": {
+           |           "emailId": "#$${$messageId}",
+           |           "envelope": {
+           |             "mailFrom": {"email": "$${$username}"},
+           |             "rcptTo": [{"email": "$${$recipient}"}]
+           |           }
+           |         }
+           |    }
+           |  }, "c1"]]
+           |}""".stripMargin
```
Comment (Arsnael) [inline src/main/scala-2.12/org/apache/james/gatling/jmap/rfc8621/JmapEmail.scala:190]: Might need to compare again with the draft scenario... Not sure we use the params defined in mailFeeder above here for example

File: src/test/scala-2.12/org/apache/james/gatling/simulation/SimulationOnMailCorpus.scala (hunk 4 above)
Comment (LanKhuat) [reply src/test/scala-2.12/org/apache/james/gatling/simulation/SimulationOnMailCorpus.scala]: What do you mean? It is the same as draft?

Comment (Arsnael) [reply src/test/scala-2.12/org/apache/james/gatling/simulation/SimulationOnMailCorpus.scala]: Not exactly... In the draft simulation, we don't depend on the provisioning (does not extend `SimulationOnMailCorpus`). We don't need the provisioning here I believe, you shouldn't need to touch this file I think

File: src/test/scala-2.12/org/apache/james/gatling/simulation/jmap/rfc8621/EmailSubmissionSimulation.scala [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package org.apache.james.gatling.simulation.jmap.rfc8621
+
+import io.gatling.core.scenario.Simulation
+import org.apache.james.gatling.jmap.MessageSubmission
+import org.apache.james.gatling.jmap.rfc8621.scenari.EmailSubmissionScenario
+import org.apache.james.gatling.simulation.{Configuration, SimulationOnMailCorpus}
+
+import scala.concurrent.duration.DurationInt
+
+class EmailSubmissionSimulation extends Simulation with SimulationOnMailCorpus {
```
Comment (Arsnael) [inline src/test/scala-2.12/org/apache/james/gatling/simulation/jmap/rfc8621/EmailSubmissionSimulation.scala]: No need to extend with `SimulationOnCorpus` here

**Review by chibenwa — APPROVED** [review]

**Comment by Arsnael** [comment]
Can you revert the changes made on `SimulationOnMailCorpus`, now that you don't use it anymore for your simulation? Thanks :)

**Review by Arsnael — APPROVED** [review]

**Pull request merged**